lnrpc/walletrpc/walletkit_server.go

```diff
@@ -1625,6 +1636,220 @@ func parseAddrType(addrType AddressType,
 	}
 }
 
+// Text used to signify that a signed message follows and to prevent
+// inadvertently signing a transaction or a signature.
+// Same as bitcoin core and btcd.
+const messageSignatureHeader = "Bitcoin Signed Message:\n"
+
+// SignMessageWithAddr signs a message with the private key of the provided
+// address. The address needs to belong to the lnd wallet.
+func (w *WalletKit) SignMessageWithAddr(ctx context.Context,
+	req *SignMessageWithAddrRequest) (
+	*SignMessageWithAddrResponse, error) {
+
+	addr, err := btcutil.DecodeAddress(req.Addr, w.cfg.ChainParams)
+	if err != nil {
+		return nil, fmt.Errorf("unable to decode address %w", err)
+	}
+
+	if !addr.IsForNet(w.cfg.ChainParams) {
+		return nil, fmt.Errorf("encoded address is for"+
+			"the wrong network %s", req.Addr)
+	}
+
+	// Fetch Address Infos from own wallet and check whether it belongs
+	// to the lnd wallet
+	managedAddr, err := w.cfg.Wallet.AddressInfo(addr)
+	if err != nil {
+		return nil, fmt.Errorf("address could not be found in the "+
+			"wallet database %w", err)
+	}
+
+	// Verify that we own the private key to the provided address
+	pubKey, ok := managedAddr.(waddrmgr.ManagedPubKeyAddress)
+	if !ok {
+		return nil, fmt.Errorf("private key to address is unknown")
+	}
+
+	// TODO: Cannot use the signer here because its restricted to the 1017
```

Comment:
I added this comment to either go with the given implementation or chain the keychain package to also add the scope of the key? So feedback needed here.

Reply:
I took this comment out, bc restructering the keychain package would increase the architectural change a lot.